src/Utilities/tests/test_rng.cpp

```diff
@@ -22,28 +21,6 @@ using std::string;
 using std::vector;
 namespace qmcplusplus
 {
-#ifdef HAVE_LIBBOOST
```

Comment:
Std::mt19937 should still be deterministic with default construction.  Or at least that's what I think specifying what the 10000 output should guarantee.

Reply:
I will add the determinism to the StdRandom test.

Reply:
Just migrated the test from BoostRandom to StdRandom. Kept testing only the first 3 elements. Even if we check 10000th and it went wrong, there won't be much we can do. So testing first 3 should be sufficient.